Jenkinsfile

```diff
@@ -33,93 +34,68 @@ metadata:
 spec:
   containers:
   - name: k8s-node
-    image: gcr.io/pso-helmsman-cicd/jenkins-k8s-node:1.1.0
-    imagePullPolicy: Always
-    command:
-    - cat
+    image: gcr.io/pso-helmsman-cicd/jenkins-k8s-node:${env.CONTAINER_VERSION}
+    command: ['cat']
     tty: true
     volumeMounts:
-    # Mount the docker.sock file so we can communicate with the local docker
-    # daemon
-    - name: docker-sock-volume
-      mountPath: /var/run/docker.sock
-    # Mount the local docker binary
-    - name: docker-bin-volume
-      mountPath: /usr/bin/docker
     # Mount the dev service account key
     - name: dev-key
       mountPath: /home/jenkins/dev
   volumes:
-  - name: docker-sock-volume
-    hostPath:
-      path: /var/run/docker.sock
-  - name: docker-bin-volume
-    hostPath:
-      path: /usr/bin/docker
   # Create a volume that contains the dev json key that was saved as a secret
   - name: dev-key
     secret:
       secretName: jenkins-deploy-dev-infra
 """
-    }
-  }
-
-  environment {
-    GOOGLE_APPLICATION_CREDENTIALS    = '/home/jenkins/dev/jenkins-deploy-dev-infra.json'
-  }
-
-
-  stages {
-    stage('Lint') {
-      steps {
-        container('k8s-node') {
-          sh "make lint"
-        }
-      }
-    }
+ ) {
+ node(label) {
+  try {
+    // set env variable GOOGLE_APPLICATION_CREDENTIALS for Terraform
+    env.GOOGLE_APPLICATION_CREDENTIALS=GOOGLE_APPLICATION_CREDENTIALS
+    // checkout code from scm i.e. commits related to the PR
+    checkout scm
 
     stage('Setup') {
-      steps {
         container('k8s-node') {
-          script {
-            env.ZONE = "${ZONE}"
-            env.PROJECT_ID = "${PROJECT_ID}"
-            env.REGION = "${REGION}"
-            env.KEYFILE = GOOGLE_APPLICATION_CREDENTIALS
-          }
           // Setup gcloud service account access
-          sh "gcloud auth activate-service-account --key-file=${env.KEYFILE}"
+          sh "gcloud auth activate-service-account --key-file=${GOOGLE_APPLICATION_CREDENTIALS}"
           sh "gcloud config set compute/zone ${env.ZONE}"
           sh "gcloud config set core/project ${env.PROJECT_ID}"
           sh "gcloud config set compute/region ${env.REGION}"
-
          }
-        }
+    }
+    stage('Lint') {
+        container('k8s-node') {
+          sh "make lint"
+      }
     }
 
     stage('Create') {
-      steps {
         container('k8s-node') {
           sh "make create"
         }
-      }
     }
 
     stage('Validate') {
-      steps {
         container('k8s-node') {
           sh "make validate"
         }
-      }
     }
-  }
 
-  post {
-    always {
+  }
+   catch (err) {
+      // if any exception occurs, mark the build as failed
+      currentBuild.result = 'FAILURE'
+      echo "FAILURE"
```

Comment:
Do we need this echo?

Reply:
I think the intent is to say specific about the failure in "Console output" but I am not sure what to say here yet.

Reply:
added error output